.github/workflows/deploy-staging.yml

```
@@ -0,0 +1,408 @@
+  # ────────────────────────────────────────────────────────────────────
+  # Build API
+  # ────────────────────────────────────────────────────────────────────
+  build-api:
+    name: "Build API"
+    needs: changes
+    if: needs.changes.outputs.api == 'true'
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: actions/setup-go@v4
```

actionlint (1.7.12)

[error] the runner of "actions/setup-go@v4" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)